renaissance-core/src/main/java/org/renaissance/core/ModuleLoader.java

```diff
@@ -351,6 +364,52 @@ private static void extractResource(String resourcePath, Path outputFilePath)
     }
   }
 
+  /** Create classloader from list of Path. */
+  private static ClassLoader createClassLoaderFromPaths(
+    Collection<Path> classPath,
+    String name
+  ) throws ModuleLoadingException {
+    URL[] classPathUrls = pathsToUrls(classPath);
+    if (logger.isLoggable(Level.CONFIG)) {
+      logger.config(String.format(
+        "Class path for %s: %s", name,
+        Arrays.stream(classPathUrls).map(Object::toString).collect(joining(","))
+      ));
+    }
+
+    if (classPathUrls.length != classPath.size()) {
+      throw new ModuleLoadingException("malformed URL(s) in classpath specification");
+    }
+
+    ClassLoader parent = ModuleLoader.class.getClassLoader();
+    return new URLClassLoader(classPathUrls, parent);
+  }
+
+  /** Read all manifests and find first one having given property.
+   * @returns Property value or null if not found.
+   */
+  private static String getManifestProperty(ClassLoader loader, String property) {
+    try {
+      Enumeration<URL> manifests = loader.getResources("META-INF/MANIFEST.MF");
+      while (manifests.hasMoreElements()) {
+        try {
+          URL manifestUrl = manifests.nextElement();
+          Properties props = new Properties();
+          InputStream manifest = manifestUrl.openStream();
+          props.load(manifest);
+          manifest.close();
+          if (props.containsKey(property)) {
+            return props.getProperty(property);
+          }
+        } catch (IOException e) {
+          continue;
+        }
+      }
+    } catch (IOException e) {
+      // Ignore.
+    }
+    return null;
+  }
```

Comment:
I would move it to the `BenchmarkSuite` class, it only supports the management decision that is being taken in the `BenchmarkSuite`. I plan to move (static) methods methods that are not essential for the module loading mechanism from the `ModuleLoader` class into the `BenchmarkSuite` class. It's not something critical, but at least in this case, I'd avoid putting more code there.

Reply:
> I would move it to the `BenchmarkSuite` class, it only supports the management decision that is being taken in the `BenchmarkSuite`.

That makes sense. However, it is used internally by the `loadDescribedExtension` as I wanted to reuse the initialized class-loader. I think that moving just this one method would not make sense without further changes. But I do not see what to move in this case unless we want to duplicate the loader initialization. Any ideas?

Reply:
> That makes sense. However, it is used internally by the `loadDescribedExtension` as I wanted to reuse the initialized class-loader.

I overlooked that bit.

> I think that moving just this one method would not make sense without further changes. But I do not see what to move in this case unless we want to duplicate the loader initialization. Any ideas?

Actually, because "extension" is a concept of the suite, I really wanted to pull all extension loading out of `ModuleLoader` and have it only provide lower-level services related to loading a class using a classloader built from a bunch of jars (which sometimes need to be unpacked) so that it is useful both to `Launcher` and `BenchmarkSuite` classes. The responsibility of the `Launcher` and the `BenchmarkSuite` is to determine the module (or the jars that make up the class path) and the class name, and ask the `ModuleLoader` to load (and possibly instantiate) the class.

Can you just move a bigger chunk, i.e., the `loadDescribedExtension` as well? The rest can come later. But if you don't feel like doing it in this change set, we can leave it for later.

Reply:
> Can you just move a bigger chunk, i.e., the `loadDescribedExtension` as well? The rest can come later. But if you don't feel like doing it in this change set, we can leave it for later.

That should include moving the plain `loadExtension` as well and perhaps something else too. I will do that in a separate PR.

Reply:
OK